commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/VarianceTest.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.statistics.descriptive;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.Arrays;
+import java.util.stream.Stream;
+import org.apache.commons.rng.UniformRandomProvider;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Test for {@link Variance}.
+ */
+final class VarianceTest {
+
+    @Test
+    void testEmpty() {
+        Variance var = Variance.create();
+        Assertions.assertEquals(Double.NaN, var.getAsDouble());
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testVariance")
+    void testVariance(double[] values) {
+        double expected = computeExpectedVariance(values);
+        Variance var = Variance.create();
+        for (double value : values) {
+            var.accept(value);
+        }
+        TestHelper.assertEquals(expected, var.getAsDouble(), 7, () -> "variance");
+        TestHelper.assertEquals(expected, Variance.of(values).getAsDouble(), 5, () -> "of (values)");
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testVariance")
+    void testParallelStream(double[] values) {
+        double expected = computeExpectedVariance(values);
+        double actual = Arrays.stream(values)
+                .parallel()
+                .collect(Variance::create, Variance::accept, Variance::combine)
+                .getAsDouble();
+        TestHelper.assertEquals(expected, actual, 7, () -> "parallel stream");
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testVariance")
+    void testVarianceRandomOrder(double[] values) {
+        UniformRandomProvider rng = TestHelper.createRNG();
+        for (int i = 1; i <= 10; i++) {
+            testVariance(TestHelper.shuffle(rng, values));
+            testParallelStream(TestHelper.shuffle(rng, values));
+        }
+    }
+
+    static Stream<Arguments> testVariance() {
```

Comment:
It may be better here to use MeanTest (since the data is the same):
```java
    static Stream<Arguments> testVariance() {
        return MeanTest.testMean();
    }
```

My preferred option would be to move the stream method to a new class: `TestData` with `static Stream<double[]> testValues()`. Note that you do not have to use JUnit `Arguments` when the test method only has one argument. You can simply stream the objects. JUnit will map the stream elements to the test argument.

Note: When I made the above change I also had the extra test case for Mean included. That fails:
```
Double.MAX_VALUE, Double.MAX_VALUE, -Double.MAX_VALUE

expected: Infinity
actual: NaN
```

This highlights that we really should reuse the same data on all the statistics, and it should be in one place.